File: LeetCode/300-LongestIncreasingSubsequence.java
// N2 solution
class Solution {
    public int lengthOfLIS(int[] nums) {
        int[] dpA = new int[nums.length];

        int index = 0;

        while(index < nums.length) {
            int subIndex = index - 1;

            dpA[index] = 1;

            while(subIndex >= 0) {
                if(nums[index] > nums[subIndex]) {
                    dpA[index] = Math.max(dpA[subIndex] + 1, dpA[index]);
                }

                subIndex--;
            }

            index++;
        }

        return Arrays.stream(dpA).max().getAsInt();
    }
}

File: LeetCode/Solutions/Python/300-LongestIncreasingSubsequence.py
class Solution(object):
    def lengthOfLIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        dpa = []

        for index in range(len(nums)):
            dpa.append(1)

            subIndex = index - 1
            while(subIndex >= 0):
                if(nums[subIndex] < nums[index]):
                    dpa[index] = max(dpa[subIndex] + 1, dpa[index])
                subIndex -= 1

        return max(dpa)